### PR TITLE
schema converter improvements

### PR DESCRIPTION
**Pull request by jkmar**
Schema converter improvements:
-added support for schema inheritance and parents
-changed command line client
-changed error handling
-fixed double tags

### CODE REVIEW

File: item/object.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+func (item *Object) Name() string {
+	return item.objectType
+}
+
+func (item *Object) Type(suffix string) string {
+	return util.ToGoName(item.Name(), suffix)
+}
+
+func (item *Object) IsObject() bool {
```
Comment (zimnx) [inline item/object.go]: Remove it, it's used only in tests and shouldn't belong to Item interface.

File: item/item.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package item
+
+import (
+	"github.com/zimnx/YamlSchemaToGoStruct/set"
+	"github.com/zimnx/YamlSchemaToGoStruct/util"
+)
+
+type Item interface {
+	Type(string) string
```
Comment (zimnx) [inline item/item.go]: comments about each method and parameter would be nice

**Review by zimnx — CHANGES_REQUESTED** [review]

**Comment by jkmar** [comment]
@zimnx @przemyslaw-dobrowolski-cl @p-kozlowski @michalzurawski
Please review.

File: item/array.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/zimnx/YamlSchemaToGoStruct/set"
+)
+
+// Array is an implementation of Item interface
+type Array struct {
+	item Item
+}
+
+// Type implementation
+func (item *Array) Type(suffix string) string {
```
Comment (michalzurawski) [inline item/array.go]: Could you please use a different name for the variable than 'item' as it is the package name?

**Review by zimnx — APPROVED** [review]

**Comment by zimnx** [comment]
Very good job, code is well designed

**Pull request merged**